### PR TITLE
chore(core): removed various deprecated things

### PR DESCRIPTION
**Pull request by jdalsem**

### CODE REVIEW

File: views/json/page/default.php [hunk 1]
```diff
@@ -11,10 +11,3 @@
 elgg_set_http_header("Content-Type: application/json;charset=utf-8");
 
 echo $vars['body'];
-
-// backward compatibility
-global $jsonexport;
```
Comment (mrclay) [inline views/json/page/default.php]: Please make sure core doesn't use this elsewhere. I think that's why I left this.

Comment (mrclay) [inline views/json/page/default.php]: #9424

Comment (jdalsem) [inline views/json/page/default.php]: its deprecated... so it can be removed in master am i right? the other ticket can take care of the leftovers

Comment (hypeJunction) [inline views/json/page/default.php]: It's deprecated without a workaround. We had to add it back because it broke webservices.

Comment (jdalsem) [inline views/json/page/default.php]: removed this removal from this PR... let #9424 fix it

**Comment by jdalsem** [comment]
resolved conflicts.. ready to merge

File: docs/guides/upgrading.rst [hunk 2]
```diff
@@ -86,7 +86,26 @@ Form and field related changes
...
+ * ``groupicon``
+ * ``twitterservice``
+
+Removed actions
+---------------
+
+ * ``file/download``: Use ``elgg_get_inline_url`` or ``elgg_get_download_url``
+ * ``import/opendd``
+ 
+ Inheritance changes
```
Comment (mrclay) [inline docs/guides/upgrading.rst]: Remove leading space here and build should pass

Comment (mrclay) [inline docs/guides/upgrading.rst]: I'm going to finish this up.

**Comment by mrclay** [comment]
Replaced by #10392

**Pull request closed without merging**